driver-dse/src/main/java/com/datastax/driver/graph/GraphSession.java

```diff
@@ -0,0 +1,225 @@
+
+import com.google.common.collect.ImmutableMap;
+
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.exceptions.InvalidQueryException;
+
+/**
+ * The GraphSession object allows to execute and prepare graph specific statements.
+ *
+ * You should use this object whenever the statement to execute is a graph query.
+ * The object will make verifications before executing queries to make sure the
+ * statement sent is valid.
+ * This object also generates graph result sets.
+ */
+public class GraphSession {
+    private final Session session;
+    private final ConcurrentMap<String, ByteBuffer> defaultGraphPayload;
+
+    // Static keys for the Custom Payload map
+    final static String GRAPH_SOURCE_KEY;
+    final static String GRAPH_LANGUAGE_KEY;
+    final static String GRAPH_KEYSPACE_KEY;
+    final static String GRAPH_REBINDING_KEY;
+
+    // Add static DefaultPayload values for Graph
+    final static String DEFAULT_GRAPH_LANGUAGE;
+    final static String DEFAULT_GRAPH_SOURCE;
+    final static Map<String, ByteBuffer> DEFAULT_GRAPH_PAYLOAD;
+
+    static {
+        GRAPH_SOURCE_KEY        = "graph-source";
+        GRAPH_LANGUAGE_KEY      = "graph-language";
+        GRAPH_KEYSPACE_KEY      = "graph-keyspace";
+        GRAPH_REBINDING_KEY     = "graph-rebinding";
+
+
+        DEFAULT_GRAPH_LANGUAGE  = "gremlin-groovy";
+        DEFAULT_GRAPH_SOURCE    = "default";
+        DEFAULT_GRAPH_PAYLOAD   = ImmutableMap.of(
+            // For the first versions of the driver Gremlin-Groovy is the default language
+            GRAPH_LANGUAGE_KEY, ByteBuffer.wrap(DEFAULT_GRAPH_LANGUAGE.getBytes()),
+
+            //If not present, the default source configured for the Keyspace
+            GRAPH_SOURCE_KEY, ByteBuffer.wrap(DEFAULT_GRAPH_SOURCE.getBytes())
+        );
+    }
+
+    /**
+     * API
+     */
+
+    /**
+     * Create a GraphSession object that wraps an underlying Java Driver Session.
+     * The Java Driver session have to be initialised.
+     *
+     * @param session the session to wrap, basically the return of a cluster.connect() call.
+     */
+    public GraphSession(Session session) {
+        this.session = session;
+        this.defaultGraphPayload = new ConcurrentHashMap<String, ByteBuffer>(DEFAULT_GRAPH_PAYLOAD);
+    }
+
+    /**
+     * Execute a graph statement.
+     *
+     * @param statement Any inherited class of {@link com.datastax.driver.graph.AbstractGraphStatement}. For now,
+     *                  this covers execution of simple and bound graph statements. It is extensible in case
+     *                  users would need to implement their own way {@link com.datastax.driver.graph.AbstractGraphStatement}.
+     * @return A {@link GraphResultSet} object if the execution of the query was successful. This method can throw an exception
+     * in case the execution did non complete successfully.
+     */
+    public GraphResultSet execute(AbstractGraphStatement statement) {
```

Comment:
I think we need an interface above `AbstractGraphStatement` or we need to make it public since this is a public method, there could be weird classloading issues (BootstrapMethodError) if someone tries to use this as a method reference in java 8.